Modules/_threadmodule.c

```diff
@@ -21,17 +20,26 @@
 // ThreadError is just an alias to PyExc_RuntimeError
 #define ThreadError PyExc_RuntimeError
 
-
 // Forward declarations
 static struct PyModuleDef thread_module;
 
+static void
+thread_run(void *boot_raw);
+struct bootstate;
+static void
+thread_bootstate_free(struct bootstate *boot, int decref);
```

Comment:
formatting:

```suggestion
static void thread_run(void *boot_raw);
struct bootstate;
static void thread_bootstate_free(struct bootstate *boot, int decref);
```

Also, it would be cleaner if we move the actual struct/function definitions up here.

Finally, let's move this to after the module state stuff rather than before.